### PR TITLE
Provsup 117

### PR DESCRIPTION
**Pull request by bri-gaia**

### CODE REVIEW

File: app/models/ca_sets.php [hunk 1]
```diff
@@ -883,6 +883,373 @@ public function getSets($pa_options=null) {
...
+	 *			publicUsers =
+	 *			name =
+	 *          byUser = return sets grouped by user with access. The array will be key'ed by sortable user name. Each entry includes a 'user' array with information about the user and a 'sets' array with the list of sets that user has access to. [Default is false]
+	 *          codes =
+	 *          includeItems =
+	 *          item_id = Get set that contains specified item_id
+	 *
+	 * @return array A list of sets keyed by set_id and then locale_id. Keys for the per-locale value array include: set_id, set_code, status, public access, owner user_id, content table_num, set type_id, set name, number of items in the set (item_count), set type name for display and set content type name for display. If setIDsOnly option is set then a simple array of set_id values is returned instead.
+	 */
+	public function getSetsBySQL($pa_options=null) {
```
Comment (kehh) [inline app/models/ca_sets.php:911]: I know this is copypasta - I'm wondering whether it's possible to change the getSets to take an argument which makes it return the sql that this function can then use / modify to add the sort / limit parameters to. Perfectly fine for you to say - nah - too hard ;)

Comment (bri-gaia) [reply app/models/ca_sets.php:911]: @kehh The issue I see with doing it that way is that there are 3(?) separate queries that would need to be returned and run in a different order depending on which way we are doing it - counts go first currently, but I need to have the paged list of items to limit the query.

I would ideally re-write the whole thing, but we discussed leaving the current function as is so we don't need to retest other parts of the system that use the original function, so I was trying to mirror as closely as possible (thinking about future upgrades primarily)

Happy to go either way on this, but ideally I'd prefer to get this up somewhere we can test to ensure it actually has the desired impact first?

**Comment by kehh** [comment]
I have no recollection of merging this :exploding_head:

**Comment by kehh** [comment]
> I have no recollection of merging this 🤯

Oh wait this isn't GitLab. I remember now.

**Comment by bri-gaia** [comment]
@kehh I have re-implemented sorts, however this did not include "set content type" or "set type" as these are both only available as ids in the query. 
We could look at:
* pulling in the label - but that would re-add time (not much)
* order by id - but this would probably lead to similar confusion as using the id to sort by 'created'

Let me know if you think I should do either/some third thing :D

One more change inc.

**Comment by kehh** [comment]
Set content type is not possible to sort in the db as that's just the table_num - table name doesn't live in the db.

Set type should be able to pull in the list item label for the type_id and wouldn't be expensive - go ahead with this if you can otherwise it's fine to exclude it from sort.

**Pull request merged**